### PR TITLE
[All hosts] (NAA) update for changes in MSAL library

### PR DESCRIPTION
**Pull request by davidchesnut**
Updates API changes for MSAL browser 3.15.0 and later.

### CODE REVIEW

File: docs/develop/enable-nested-app-authentication-in-your-add-in.md [hunk 1]
```diff
@@ -61,25 +61,25 @@ The following steps show how to enable NAA in the `taskpane.js` or `taskpane.ts`
...
         document.getElementById("sideload-msg").style.display = "none";
         document.getElementById("app-body").style.display = "flex";
         document.getElementById("run").onclick = run;
   
         // Initialize the public client application
-        pca = await PublicClientNext.createPublicClientApplication({
+        pca = await createNestablePublicClientApplication({
           auth: {
             clientId: "Enter_the_Application_Id_Here",
             authority: "https://login.microsoftonline.com/common",
```
Comment (codexeon) [inline docs/develop/enable-nested-app-authentication-in-your-add-in.md:85]: supportsNestedAppAuth is no longer necessary after switching to createNestablePublicClientApplication. We should remove it from documentation

**Comment by codexeon** [comment]
supportsNestedAppAuth is mentioned a few places, we should explain what createNestablePublicClientApplication does without use of supportsNestedAppAuth since it is a deprecated config setting.

File: docs/develop/enable-nested-app-authentication-in-your-add-in.md [hunk 2]
```diff
@@ -61,25 +61,25 @@ The following steps show how to enable NAA in the `taskpane.js` or `taskpane.ts`
...
 Next, you need to initialize MSAL and get an instance of the public client application. This is used to get access tokens when needed. We recommend that you put the code that creates the public client application in the `Office.onReady` method.
 
-- In your `Office.onReady` function, add a call to `createPublicClientApplication` as shown below. Replace the `Enter_the_Application_Id_Here` placeholder with the Azure app ID you saved previously.
+- In your `Office.onReady` function, add a call to `createNestablePublicClientApplication` as shown below. Replace the `Enter_the_Application_Id_Here` placeholder with the Azure app ID you saved previously.
 
     ```javascript
     let pca = undefined;
     Office.onReady(async (info) => {
-      if (info.host === Office.HostType.Excel) {
+      if (info.host) {
```
Comment (codexeon) [inline docs/develop/enable-nested-app-authentication-in-your-add-in.md:76]: should info.host just be removed? I don't see it referenced anywhere.

Comment (davidchesnut) [reply docs/develop/enable-nested-app-authentication-in-your-add-in.md:76]: It's a part of yo office that checks that you are sideloaded. If so there is a host. If not, the task pane will inform you to sideload. So I need to check for this to ensure the task pane updates correctly. However this code will run in any host (not just Excel) so no need to check which host is used.

**Pull request merged**